### PR TITLE
Set all transforms to train during validation

### PR DESCRIPTION
**Pull request by KnathanM**
When `lightning` runs the validation loop, it automatically sets the model to `eval` mode. This is needed to turn off dropout and to tell the batch normalization layer to use its running average statistics instead of batch statistics, the same as is done during inference. 

But this also sets the `output_transform` to `eval`, which would cause the model to unscale the model predictions before comparing with the target values stored in the validation dataset. This is a problem because we made the choice to keep the validation loss in the same scaled space as the training loss so that they would be some what comparable. To do this we scale the target values in the validation dataset, so they should be compared to the raw scaled-space predictions of the model (not the unscaled version). We fixed this by manually setting the `output_transform` back to `train` in `on_validation_model_eval`.

Our examples also show that the graph transforms and extra datapoint descriptors transforms should be scaled in the validation set, based on the scaling in the training dataset. This isn't actually necessary, but what we have chosen. Currently though we don't set those transforms to training mode so the extra descriptors are being double scaled. I have fixed that.

This also makes me wonder why we do any scaling in the datasets themselves. Couldn't we just fit a scaler to the training data, add it to the model and then always perform the transform scaling? The train and val loss would both be in the unscaled space and maybe that is less preferred?

### CODE REVIEW

**Comment by KnathanM** [comment]
I checked out v2.0.3 and the bug with scaling features/descriptors was there too. The symptom of the bug is that during training the validation loss is best at the first epoch and then rises. This is a problem if the "best" model is chosen based on the validation loss. This doesn't affect the actual training and we probably didn't see this earlier because we often use the last model from training for prediction, or haven't used extra data descriptors.

**Review by shihchengli — APPROVED** [review]
Thanks for the fix. I think it is a good idea to not scale targets in the datasets themselves directly. Instead, using the existing `output_transform` in predictor to transform the targets before calculating the metrics. It would be an additional step for the scaling operation in each training epoch, but I don't expect it would increase the computational time significantly. I think you can still return the scaled loss in training/validation if you design some functions in the `MPNN` class (e.g., `on_validation_model_eval`) properly.

**Review by JacksonBurns — APPROVED** [review]

**Comment by AsaBarthMaron** [comment]
I am having a loosely related issue. When building regression models, the val metrics (e.g., MAE, RMSE) are in the incorrect units, since they are being calculated in the scaled space. 

I have decided to make a change to my local code to 

- comment out the `*_transform.train()` lines in `on_validation_model_eval()` (so all the method does is set `self.eval()`) 
- comment out the `if self.training:` statements in `transforms.py`
- obtain an `output_scaler` object directly, without calling .transform() on the train or val datasets

I seem to be getting the same/very similar results as prior to making these changes, except that the val metrics are now in the right units. 

If having the scaling be handled exclusively by the output transform and not the datasets is a direction you are considering going, I would greatly appreciate similar changes be made incorporated in a future commit/version!

**Comment by JacksonBurns** [comment]
Hi @AsaBarthMaron I am going to convert your comment to an issue for better visibility
